config/overrides.go

```diff
@@ -55,6 +55,8 @@ var groupMap = map[string]GroupKindCalculator{
 	"google_project_usage_export_bucket$":          ReplaceGroupWords("cloudplatform", 0),
 	"google_service_account.*":                     ReplaceGroupWords("cloudplatform", 0),
 	"google_service_networking_peered_dns_domain$": ReplaceGroupWords("cloudplatform", 0),
+	// Avoid an import cycle, this is a key used for S3-style auth. It's meant to be used by end users, not other resources.
+	"google_storage_hmac_key$": ReplaceGroupWords("cloudplatform", 0),
```

Comment:
I verified that if we don't make a group change it causes an import cycle:
<img width="1269" alt="Screenshot 2023-06-22 at 23 20 43" src="https://github.com/upbound/provider-gcp/assets/103541666/45e6d791-8d79-4ddf-9118-66edaf5f5390">

But I'm not sure if this is the right solution here. @ulucinar, @sergenyalcin what do you think about this?

Reply:
@turkenf as far as I understand, the reason of this import cycle is reference addition in the config file.

In the context of API groups, I don't think this is the right solution approach, since we depend on external providers' groups.

Until today, we have chosen the method of removing references in these cases. I am aware that this is not an ideal solution either, but in order to solve this problem in the long term, it is necessary to address it at the upjet level.

Reply:
> @turkenf as far as I understand, the reason of this import cycle is reference addition in the config file.

@sergenyalcin in fact, even if no reference is defined in the config file, we encounter an import cycle error because the same resource is automatically injected with references.

We can reset it with the following configuration in `config/storage/config.go` and in this case, we do not encounter an import cycle error:

```
	p.AddResourceConfigurator("google_storage_hmac_key", func(r *config.Resource) {
		r.References["service_account_email"] = config.Reference{}
	})
```

Should we proceed this way?